fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ReduceCastRule.java

```diff
@@ -209,11 +197,14 @@ public ScalarOperator reduceDateToDatetimeCast(BinaryPredicateOperator operator)
         return resultBinaryPredicateOperator;
     }
 
-    public ScalarOperator reduceDatetimeToDateCast(BinaryPredicateOperator operator) {
-        ScalarOperator child1 = operator.getChild(0);
-        ScalarOperator child2 = operator.getChild(1);
-        ScalarOperator castChild = child1.getChild(0);
-        LocalDateTime originalDate = ((ConstantOperator) child2).getDate();
+    private ScalarOperator reduceDatetimeToDateCast(BinaryPredicateOperator operator) {
+        ScalarOperator castChild = operator.getChild(0).getChild(0);
+        ConstantOperator child2 = (ConstantOperator) operator.getChild(1);
+        if (child2.isNull()) {
+            return operator;
+        }
+
+        LocalDateTime originalDate = child2.getDate().truncatedTo(ChronoUnit.DAYS);
```

Comment:
Why is bottomDateTime.truncatedTo(ChronoUnit.DAYS) needed here, and when we need to call this function

Reply:
I think you should read the comment about the rule